mw_jwt.go

```diff
@@ -100,31 +108,41 @@ func (k *JWTMiddleware) getSecretFromURL(url, kid, keyType string) ([]byte, erro
 	return nil, errors.New("No matching KID could be found")
 }
 
-func (k *JWTMiddleware) getIdentityFomToken(token *jwt.Token) (string, bool) {
-	// Try using a kid or sub header
+// Try using a kid or configured claim or sub claim
+func (k *JWTMiddleware) getIdentityFomToken(token *jwt.Token) (string, bool, error) {
 	idFound := false
 	var tykId string
-	if token.Header["kid"] != nil {
-		tykId = token.Header["kid"].(string)
-		idFound = true
+
+	// Global config overrides and sets the api def
+	if k.Spec.GlobalConfig.JWTSkipCheckKidAsId {
+		k.Spec.APIDefinition.JWTSkipCheckKidAsId = true
```

Comment:
Actually seemed trivial to me. But if I want it on all my apis, because my jwts always contain kid (that I'm using for something else or cannot control it) then this way I can make sure it's always ON. I will not need to keep on setting it every time or worse - will forget to do that and then the jwt will fail and I'll have to update all my apis. 
Also the case of using kid for something else would probably be always applied to all my apis since I'm using the same IdP and same rules.

Reply:
but it doesn't give you option to make sure it is always OFF for all your apis - thats confusing for me as well